### DISCUSSION TITLE
Possible issue with admin token when using reverse proxy?

### DISCUSSION THREAD

**Discussion by seriousm4x** [pocketbase/pocketbase#3170] [Q&A]
Hey, first of all, thank you for the support you provide here. That's not taken for granted.

I currently have 2 users in my project who are experiencing a very bizarre and rare issue with their reverse proxy. I'm not even 100% sure if it's an issue with Pocketbase, but I have a strong suspicion.

* [The first user](https://github.com/seriousm4x/UpSnap/issues/65#issue-1640500886) encountered the error `403: Only admins can perform this action` when using my dashboard via nginx proxy manager. Initially, I thought the error might be somewhere in my code or related to his reverse proxy setup and I hadn't considered that it might be a problem with Pocketbase.

* Now, however, [a second user](https://github.com/seriousm4x/UpSnap/issues/65#issuecomment-1681354339) has reported the exact same issue (using SWAG, also nginx). He also received a 403 error when accessing the page through a reverse proxy, but everything works fine when using the direct IP.

The reason I'm writing here in the discussions is that the second user pointed out that **the same problem occurs with the Pocketbase backend at `/_/`**. There, he also receive the same error message. So maybe the issue isn't on my end?

I've tried to reproduce this issue multiple times, but everything seems to be working fine on my end. I just wanted to ask if there are any known issues related to this or if you have any suggestions for a possible solution.

I know it's a fairly long issue and it's not your project, but I suspect it might be an issue with pocketbase somehow.

I'd be grateful if you could provide us with any hints on what the issue might be!
Thanks!

**Comment by ganigeorgiev** [comment]
Without concrete reproducible steps, I'm not sure how to help.

If the `Authorization` header is proxied properly, then I don't think the reverse proxy has anything to do with the admin token being invalid.

**There are no sessions and the PocketBase API is completely stateless, so as long as the request has `Authorization:TOKEN` header with it, it should work fine.**

**Reply by seriousm4x** [reply]
Thanks, will ask again for better steps. Will report back if it's actually an issue with pb.

**Reply by ganigeorgiev** [reply]
The only thing I guess could mess up the admin authentication, excluding expired token, is if the users have started different applications using the same host address because in this case the localStorage entry will be shared for both. Aka. the following scenario will fail:

- start App1 on localhost:8090
- authenticate as admin
- kill App1
- start App2 (with different pb_data) on the same localhost:8090
- since a localStorage entry for the same address exists, the store will be automatically populated with it, but the token will be invalid because its secret originate from App1 and not App2

**Reply by seriousm4x** [reply]
Hey again. So I got a detailed explanation on how to reproduce it. https://github.com/seriousm4x/UpSnap/issues/65#issuecomment-1687708031

It's definitely an issue with pocketbase itself as he can't login to the admin dashboard as well (as you can see further down). I haven't tried myself as I'm on vacation.

I have asked him to create a separate issue over here as well.

**Reply by ganigeorgiev** [reply]
@seriousm4x I've skimmed to the reproducible steps, but I again I don't see what could be the cause of it as there are too many components in the setup.

Also the comment author mention that they created a "user", but keep in mind that **PocketBase admins and users are 2 different entities** and their generated JWTs have different secrets and permissions granted.

If you think that it is a PocketBase issue, please create a **minimal** reproducible repo and I'll have a more detailed look, but as it is right now I just don't know where to look as I'm not familiar with a lot of tools used in the example.
